packages/datadog-instrumentations/src/mocha.js

```diff
@@ -19,7 +19,31 @@ const patched = new WeakSet()
 
 const testToAr = new WeakMap()
 const originalFns = new WeakMap()
-const testSuiteToAr = new WeakMap()
+const testFileToSuiteAr = new Map()
+
+function getSuitesByTestFile (root) {
+  const suitesByTestFile = {}
+  function getSuites (suite) {
+    if (suite.file) {
+      if (suitesByTestFile[suite.file]) {
+        suitesByTestFile[suite.file].push(suite)
```

Comment:
For the counter, what I meant is to explicitly use a separate counter to avoid having to filter or reduce every time the count needs to be checked. For example:

```js
suitesByTestFile[suite.file].push(suite)
activeSuiteCount++
```

And then this would be decremented as suites are being finished. With the suite property now removed, this is just an optimization though so I'll leave it up to you to do it in this PR or later if needed.